res/controllers/Roland_DJ-505-scripts.js

```diff
@@ -1405,6 +1412,53 @@ DJ505.CueLoopMode = function(deck, offset) {
 };
 DJ505.CueLoopMode.prototype = Object.create(components.ComponentContainer.prototype);
 
+DJ505.EditMode = function(deck, offset) {
+    components.ComponentContainer.call(this);
+    this.ledControl = DJ505.PadMode.HOTCUE;
+    this.color = DJ505.PadColor.ORANGE;
+
+    this.pads = new components.ComponentContainer();
+    for (var i = 0; i <= 3; i++) {
+        var baseKey = ((i < 2) ? "intro" : "outro") + "_" + ((i % 2) ? "end" : "start");
```

Comment:
I find this very hard to read. A more readable, predictable and versatile approach would be if you were to look it up instead:
```suggestion
        var baseKey = [
            "intro_start",
            "intro_end",
            "outro_start",
            "outro_end",
        ][i];
```